### PR TITLE
Refactoring addressing the comments from @ericsoderberghp

### PR DESCRIPTION
**Pull request by TiagoJustino**
- Using <a> instead of <div> on SkipLinksAnchor (@ericsoderberghp suggestion)
- SkipLinkAnchor moved from Box to Article/Section (@ericsoderberghp suggestion)
- Added tabIndex -1 to SkipLinksAnchors as suggested in [A Skip Link Primer](http://viget.com/inspire/skip-link-primer)
- Using .focus() when moving to the anchor as suggested in [A Skip Link Primer](http://viget.com/inspire/skip-link-primer)
- Removed trailing white space from scss file.

### CODE REVIEW

File: src/js/components/Article.js [hunk 1]
```diff
@@ -136,7 +136,7 @@ var Article = React.createClass({
 
     var skipLinkAnchor = null;
     if (this.props.primary) {
-      skipLinkAnchor = <SkipLinkAnchor label="Footer" />;
+      skipLinkAnchor = <SkipLinkAnchor label="Main Content" />;
```
Comment (alansouzati) [inline src/js/components/Article.js]: Thanks for that! My Fault :)

File: src/js/components/Section.js [hunk 2]
```diff
@@ -2,12 +2,16 @@
...
+var merge = require('lodash/object/merge');
 
 var CLASS_ROOT = "section";
 
 var Section = React.createClass({
 
-  propTypes: Box.propTypes,
+  propTypes: merge({
+    primary: React.PropTypes.bool
+  }, Box.propTypes),
```
Comment (alansouzati) [inline src/js/components/Section.js]: I feel we need to flip this order for the arguments. Section propTypes should take over Box propTypes.

**Comment by alansouzati** [comment]
I general looks good to me. I will wait @ericsoderberghp feedbacks on it.

File: src/js/components/Section.js (hunk 2 above)
Comment (TiagoJustino) [inline src/js/components/Section.js]: Thanks for clarifying that. =)

File: src/js/components/Section.js [hunk 3]
```diff
@@ -2,12 +2,16 @@
...
+var SkipLinkAnchor = require('./SkipLinkAnchor');
+var merge = require('lodash/object/merge');
 
 var CLASS_ROOT = "section";
 
 var Section = React.createClass({
 
-  propTypes: Box.propTypes,
+  propTypes: merge(Box.propTypes, {
+    primary: React.PropTypes.bool
```
Comment (ericsoderberghp) [inline src/js/components/Section.js]: What is the context where we would have a primary Section without a primary Article?

File: src/js/components/SkipLinks.js [hunk 4]
```diff
@@ -45,6 +45,14 @@ var SkipLinks = React.createClass({
     }.bind(this));
   },
 
+  _onClick: function (destId) {
+    return function (event) {
+      var _document = React.findDOMNode(this).ownerDocument;
```
Comment (ericsoderberghp) [inline src/js/components/SkipLinks.js]: Why can't you just use document.querySelectorAll()?

**Comment by ericsoderberghp** [comment]
Have you tested these changes across the Grommet site, medium-app, people-finder, etc.?

File: src/js/components/Section.js (hunk 3 above)
Comment (TiagoJustino) [inline src/js/components/Section.js]: Our Docs project is an example. On the main page we want the Section to be the main content, not the article, so we can skip the Design and Develop links on the top right corner. The same for the the Design and Develop pages.

File: src/js/components/SkipLinks.js (hunk 4 above)
Comment (TiagoJustino) [inline src/js/components/SkipLinks.js]: I can. I'll fix this.=)

**Comment by TiagoJustino** [comment]
We still need to add the primary attribute to the other apps when appropriate, as well as the others pages on the Docs. The main page of Docs is just an example on how to use it.

File: src/js/components/SkipLinks.js (hunk 4 above)
Comment (ericsoderberghp) [inline src/js/components/SkipLinks.js]: just waiting for this change

Comment (TiagoJustino) [inline src/js/components/SkipLinks.js]: Just updated. Sorry for the delay.

File: src/js/components/SkipLinks.js [hunk 5]
```diff
@@ -36,15 +35,21 @@ var SkipLinks = React.createClass({
...
+      var activeElement = document.activeElement;
       if (!DOM.isDescendant(skipLinksLayer, activeElement)) {
         this.setState({showLayer: false});
       }
     }.bind(this));
   },
 
+  _onClick: function (destId) {
+    return function (event) {
+      var dest = document.querySelector('#' + destId);
```
Comment (ericsoderberghp) [inline src/js/components/SkipLinks.js]: Sorry, I didn't notice this before. I think document.getElementById(destId) is more efficient than querySelectorAll() when you already have an id.

Comment (TiagoJustino) [inline src/js/components/SkipLinks.js]: Good to know. Thank you for pointing that out. =)

**Pull request merged**